lib/algolia/index.rb

```diff
@@ -405,13 +405,28 @@ def delete_by_query(query, params = nil)
       params.delete('attributesToRetrieve')
 
       params[:hitsPerPage] = 1000
+      params[:distinct] = false
       params[:attributesToRetrieve] = ['objectID']
+      last_task = nil
       loop do
         res = search(query, params)
         break if res['hits'].empty?
-        res = delete_objects(res['hits'].map { |h| h['objectID'] })
-        wait_task res['taskID']
+        last_task = delete_objects(res['hits'].map { |h| h['objectID'] })
```

Comment:
Isn't it introducing duplicate requests? cf. convo https://github.com/algolia/algoliasearch-client-ruby/pull/211